python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/traffic-policies.j2

```diff
@@ -7,207 +7,200 @@
-         {{ protocol_cli }}
-{#                             -- destination field-set -- #}
-{%                             elif protocol.dst_field is arista.avd.defined and protocol.protocol | lower in ['tcp','udp'] %}
-{%                                 set protocol_cli = 'protocol ' ~ protocol.protocol %}
-{%                                 if protocol.flags is arista.avd.defined %}
-{%                                     set protocol_cli = protocol_cli ~ ' flags ' ~  protocol.flags | join(' ') %}
-{%                                 endif %}
-{%                                 set protocol_cli = protocol_cli ~ ' destination port field-set ' ~ protocol.dst_field %}
-         {{ protocol_cli }}
-{#                             -- source field-set -- #}
+         {{ protocol_cli }} source port {{ protocol.src_port }}
+{%                             elif protocol.dst_port is arista.avd.defined and protocol.protocol | lower in ['tcp','udp'] %}
+         {{ protocol_cli }} destination port {{ protocol.dst_port }}
+{%                             endif %}
+{#                             -- source/destination field-set -- #}
+{%                             if protocol.src_field is arista.avd.defined and protocol.dst_field is arista.avd.defined and protocol.protocol | lower in ['tcp','udp'] %}
+         {{ protocol_cli }} source port field-set {{ protocol.src_field }} destination port field-set {{ protocol.dst_field }}
 {%                             elif protocol.src_field is arista.avd.defined and protocol.protocol | lower in ['tcp','udp'] %}
-{%                                 set protocol_cli = 'protocol ' ~ protocol.protocol %}
-{%                                 if protocol.flags is arista.avd.defined %}
-{%                                     set protocol_cli = protocol_cli ~ ' flags ' ~  protocol.flags | join(' ') %}
-{%                                 endif %}
-{%                                 set protocol_cli = protocol_cli ~ ' source port field-set ' ~ protocol.src_field %}
+         {{ protocol_cli }} source port field-set {{ protocol.src_field }}
+{%                             elif protocol.dst_field is arista.avd.defined and protocol.protocol | lower in ['tcp','udp'] %}
+         {{ protocol_cli }} destination port field-set {{ protocol.dst_field }}
+{%                             endif %}
+{%                             if protocol.icmp_type is arista.avd.defined and protocol.protocol | lower == 'icmp' %}
+         {{ protocol_cli }} type {{ protocol.icmp_type | arista.avd.natural_sort | join(' ') }} code all
+{%                             endif %}
+{%                             if protocol.protocol is arista.avd.defined and protocol.icmp_type is not arista.avd.defined and protocol.dst_field is not arista.avd.defined and protocol.src_field is not arista.avd.defined and protocol.dst_port is not arista.avd.defined and protocol.src_port is not arista.avd.defined and protocol.flags is not arista.avd.defined %}
```

Comment:
This smells like something that should be in an `else` block or checking for the specific options that are not caught above. We should not have to do negative checks on all the other settings.

Reply:
updated the logic